checks/collector.py

```diff
@@ -751,7 +751,7 @@ def _get_hostname_metadata(self):
         except Exception:
             pass
 
-        metadata["hostname"] = get_hostname()
+        metadata["hostname"] = get_hostname(config=self.agentConfig)
```

Comment:
re-reading this, it looks like that it's exactly the same call that we do before when launching the agent, see https://github.com/DataDog/dd-agent/blob/quentin/fix-nosetests-output/agent.py#L109
maybe use this instead of another call?